pyxem/generators/displacement_gradient_tensor_generator.py

```diff
@@ -104,25 +125,34 @@ def get_single_DisplacementGradientTensor(Vs, Vu=None, weights=None):
     get_DisplacementGradientMap()
 
     """
+    is_row_nan = np.logical_not(np.any(np.isnan(Vs), axis=1))
+    Vs = Vs[is_row_nan]
+    Vu = Vu[is_row_nan]
+
     if Vu is not None:
         if Vu.dtype == object:
             Vu = Vu[()]
     if weights is not None:
         # see https://stackoverflow.com/questions/27128688
         weights = np.asarray(weights)
         # Need vectors normalized to the unstrained region otherwise the weighting breaks down
-        Vs = (
-            np.divide(Vs, np.linalg.norm(Vu, axis=0)) * np.sqrt(weights)
-        ).T  # transpose for conventions
-        Vu = (np.divide(Vu, np.linalg.norm(Vu, axis=0)) * np.sqrt(weights)).T
+        norm = np.linalg.norm(Vu, axis=1)
+        Vs = (np.divide(Vs.T,
+                        np.linalg.norm(Vu, axis=1)) *
+              np.sqrt(weights)).T
+        Vu = (np.divide(Vu.T,
+                        np.linalg.norm(Vu, axis=1)) *
+              np.sqrt(weights)).T
     else:
-        Vs, Vu = Vs.T, Vu.T
+        Vs, Vu = Vs, Vu
 
-    L = np.linalg.lstsq(Vu, Vs, rcond=-1)[
-        0
-    ]  # only need the return array, see np,linalg.lstsq doc
+    L, residuals, rank, s = np.linalg.lstsq(Vu, Vs, rcond=-1)
+    # only need the return array, see np,linalg.lstsq doc
     # Put caculated matrix values into 3 x 3 matrix to be returned.
     D = np.eye(3)
     D[0:2, 0:2] = L
 
-    return D
+    if return_residuals:
+        return residuals
```

Comment:
I think this should return a tuple, `(D,residuals)`

Reply:
https://github.com/dask/dask/issues/9510

Reply:
I do need to run black on this as well.  I'm pretty bad at remembering that sometimes.  That might be something that is good to include with CI.  There probably is a way to run black before every PR for the files changed.

Reply:
Happy to let this be if it's a dask problem. We could run black for the CI (either to auto-black the code or to fail unblacked code but I think it's probably overkill). Just like to mention it to experienced devs when I see it.